src/validation.cpp

```diff
@@ -2751,7 +2751,13 @@ bool CChainState::ActivateBestChain(CValidationState &state, const CChainParams&
     CheckBlockIndex(chainparams.GetConsensus());
 
     // Write changes periodically to disk, after relay.
-    if (!FlushStateToDisk(chainparams, state, FlushStateMode::PERIODIC)) {
+    // Unless we just finished initial sync, then always write to disk
+    static bool initially_in_ibd = true;
```

Comment:
I think that this variable should be the member of CChainState to avoid share same variable if we have two CChainState object in the program.

Reply:
Done.